clang/include/clang/Basic/Attr.td

```diff
@@ -1460,6 +1460,35 @@ def SYCLIntelLoopFuse : InheritableAttr {
   let SupportsNonconformingLambdaSyntax = 1;
 }
 
+def SYCLAddIRAttributesFunction : InheritableAttr {
+  let Spellings = [CXX11<"__sycl_detail__", "add_ir_attributes_function">];
+  let Args = [VariadicExprArgument<"Args">];
+  let LangOpts = [SYCLIsDevice];
```

Comment:
All examples in the [design document](https://github.com/intel/llvm/blob/sycl/sycl/doc/CompileTimeProperties.md) for using these are wrapped in `__SYCL_DEVICE_ONLY__`. That said, I don't see anywhere saying it should emit a diagnostic otherwise, so I've added these!